spec/models/robots/robot_spec.rb

```diff
@@ -323,19 +323,218 @@
+            }
+          end
+
+          let(:target_tube_parents) { [source_plate, phix_tube] }
+
+          let(:source_plate_2) do
+            create :v2_plate,
+                   barcode_number: 5,
+                   purpose_name: source_purpose_name,
+                   state: source_plate_state
+          end
+          let(:source_plate_2_barcode) { source_plate_2.human_barcode }
+
+          let(:target_tube_2_parents) { [source_plate_2, phix_tube] }
+          let(:target_tube_2) do
+            create :v2_tube,
+                   purpose_name: target_tube_purpose_name,
+                   barcode_number: 6,
+                   state: target_tube_state,
+                   parents: target_tube_2_parents
+          end
+          let(:target_tube_2_barcode) { target_tube_2.human_barcode }
+
+          before do
+            bed_labware_lookup(source_plate)
+            bed_labware_lookup(target_tube)
+            bed_labware_lookup(source_plate_2)
+            bed_labware_lookup(target_tube_2)
+            bed_labware_lookup(phix_tube)
+          end
+
+          context 'with a valid layout' do
+            let(:scanned_layout) do
+              {
+                'bed1_barcode' => [source_plate_barcode],
+                'bed2_barcode' => [target_tube_barcode],
+                'bed3_barcode' => [source_plate_2_barcode],
+                'bed4_barcode' => [target_tube_2_barcode],
+                'bed5_barcode' => [phix_tube_barcode]
+              }
+            end
+
+            context 'and related labwares' do
+              it { is_expected.to be_valid }
+            end
+          end
+
+          context 'with an invalid layout' do
+            let(:scanned_layout) do
+              {
+                'bed1_barcode' => [source_plate_barcode],
+                'bed2_barcode' => [target_tube_barcode],
+                'bed3_barcode' => [source_plate_2_barcode],
+                'bed4_barcode' => [],
+                'bed5_barcode' => [phix_tube_barcode]
+              }
+            end
+
+            it { is_expected.not_to be_valid }
+          end
+
+          context 'with an invalid layout' do
+            let(:scanned_layout) do
+              {
+                'bed1_barcode' => [source_plate_barcode],
+                'bed2_barcode' => [target_tube_barcode],
+                'bed3_barcode' => [],
+                'bed4_barcode' => [],
+                'bed5_barcode' => [phix_tube_barcode]
+              }
+            end
+
+            it { is_expected.to be_valid }
+          end
+        end
+
+        # context 'an multiple source purposes' do
+        #   let(:source_purpose) { ['Limber Cherrypicked', 'Other'] }
+        #   let(:target_tube_parents) { [source_plate] }
+        #   it { is_expected.to be_valid }
+
+        #   context 'but of the wrong purpose' do
+        #     let(:source_purpose_name) { 'Invalid plate purpose' }
+        #     it { is_expected.not_to be_valid }
+        #   end
+        # end
```

Comment:
Did you mean to leave in this commented out test?

Reply:
I did not. Removed, thx.